src/transformers/trainer.py

```diff
@@ -1401,6 +1401,16 @@ def _inner_training_loop(
 
         model = self._wrap_model(self.model_wrapped)
 
+        if is_sagemaker_mp_enabled() and resume_from_checkpoint is not None:
+            if os.path.isfile(os.path.join(resume_from_checkpoint, WEIGHTS_NAME)):
+                state_dict = torch.load(os.path.join(resume_from_checkpoint, WEIGHTS_NAME))
+                model.load_state_dict(state_dict)
+                # release memory
+                del state_dict
+            else:
+                # We load the sharded checkpoint
+                load_sharded_checkpoint(model, resume_from_checkpoint)
```

Comment:
Instead of rewriting the content of `_load_from_checkpoint`, we could add a `model` argument to that internal function (defaults to None in which case `self.model` is used) and call it here for the sagemaker mp case.

Reply:
Added `model` as an argument to `_load_from_checkpoint`. With SMP, checkpoint can only be loaded after model is wrapped.